### ISSUE TITLE
Custom AppBar shadowColor

### ISSUE THREAD

**Issue by v7lin** [flutter/flutter#40079]

**Comment by amanv8060** [comment]
@HansMuller I think this be closed as done in #58708

**Comment by danagbemava-nc** [comment]
@amanv8060 is right, we can set the AppBar shadow color in both the `AppBarTheme` and in the `AppBar` directly. It looks like this has been in the framework for quite some time. Since [1.20.0-1.0.pre](https://github.com/flutter/flutter/releases/tag/1.20.0-1.0.pre)


<details>
<summary>screenshots</summary>

| in AppBar | in AppBarTheme | 
| -- | -- |
| <img width="437" alt="Screenshot 2022-03-23 at 12 47 21" src="https://user-images.githubusercontent.com/88313112/159703444-d8ddc41f-2a79-4082-b0b2-984ef83ccb3c.png"> | <img width="437" alt="Screenshot 2022-03-23 at 12 46 53" src="https://user-images.githubusercontent.com/88313112/159703452-ed50c52f-b2af-4b00-a738-e8fe797bae97.png"> |

</details>

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        shadowColor: Colors.pink,
        // appBarTheme: AppBarTheme(
        //   shadowColor: Colors.red,
        // ) // todo: uncomment to set the shadow color for all appbars
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        elevation: 4.0,
        title: Text(widget.title),
        shadowColor: Colors.green, //todo: uncomment to set shadow color for only this appbar
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}

```
</details>
